packages/intrepid2/src/Orientation/Intrepid2_OrientationToolsDefMatrixData.hpp

```diff
@@ -56,36 +56,53 @@
 namespace Intrepid2 {
 
   template<typename SpT>
-  template<typename BasisPtrType>
+  template<typename BasisType>
   typename OrientationTools<SpT>::CoeffMatrixDataViewType
-  OrientationTools<SpT>::createCoeffMatrixInternal(BasisPtrType basis) {
+  OrientationTools<SpT>::createCoeffMatrixInternal(const BasisType* basis) {
     const ordinal_type order(basis->getDegree());
     const std::string name(basis->getName());
     CoeffMatrixDataViewType matData;
 
+    using ExecutionSpace  = typename BasisType::ExecutionSpace;
+    using OutputValueType = typename BasisType::OutputValueType;
+    using PointValueType  = typename BasisType::PointValueType;
+
+    using hostExecutionSpace = Kokkos::DefaultHostExecutionSpace;
+
     auto ordinalToTag = basis->getAllDofTags();
     auto tagToOrdinal = basis->getAllDofOrdinal();
 
     //
     // High order HGRAD Elements
     //
 
-    if (name == "Intrepid2_HGRAD_QUAD_Cn_FEM") {
+    //std::cout << "Name: " << name << " " << order << std::endl;
+
+    if (name.find("HGRAD_QUAD") != std::string::npos) {
```

Comment:
@mperego Could you instead check getBaseCellTopology() and getFunctionSpace()?  I think this would be cleaner than the string comparison, and it also might be that my proposed string manipulations to define the names for tensor bases would end up including "HGRAD_QUAD" as a substring for say, the HGRAD_HEX basis.

Reply:
Sure, I can. I didn't realize that those methods were implemented also for the old basis functions